src/maker_protocol.rs

```diff
@@ -31,19 +29,43 @@ use crate::messages::{
 };
 use crate::wallet_sync::{CoreAddressLabelType, Wallet, WalletSwapCoin};
 
-//TODO
-//this using of strings to indicate allowed methods doesnt fit aesthetically
-//with the using of structs and serde as in messages.rs
-//there is also no additional checking by the compiler
-//ideally this array and the other strings in this file would instead be
-//structs, however i havent had time to figure out if rust can do this
-pub const NEWLY_CONNECTED_TAKER_ALLOWED_METHODS: [&str; 5] = [
-    "giveoffer",
-    "signsenderscontracttx",
-    "proofoffunding",
-    "signreceiverscontracttx",
-    "hashpreimage",
-];
+// A structure denoting expectation of type of taker message.
+// Used in the [ConnectionState] structure.
+//
+// If the recieved message doesn't match expected method,
+// protocol error will be returned.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum ExpectedMessage {
+    // Expecting Hello from a new taker
+    TakerHello,
+
+    // A Newly connected Taker is allowed
+    // send multiple messages:
+    // - GiveOffer
+    // - SignSendersContractTx
+    // - ProofOfFunding
+    // - SignReceiversContractTx
+    // - HashPreimage
+    NewlyConnectedTaker,
+
+    // Sign sender's contract
```

Comment:
Many of these comments are useless because they just repeat the name of the enum. The name of the enum was chosen carefully to make the code self-documenting and so comments are unnecessary. Consider removing these comments.